File: scala/sublist/src/main/scala/Sublist.scala
object Sublist extends Enumeration {
  type Sublist = Value
  val Equal, Unequal, Sublist, Superlist = Value

  def sublist[T](a: List[T], b: List[T]): Sublist = {
    val isSublist = isContainedBy(a, b)
    val isSuperlist = isContainedBy(b, a)

    (isSublist, isSuperlist) match {
      case (true, true) => Equal
      case (true, false) => Sublist
      case (false, true) => Superlist
      case (false, false) => Unequal
    }
  }

  private def isContainedBy[T](a: List[T], b: List[T]): Boolean = b match {
    case _ if a == Nil => true
    case _ if b.length < a.length => false
    case _ => b.sliding(a.length, 1).contains(a)
  }
}

File: swift/sublist/Sources/Sublist/Sublist.swift
enum Classifier {
  case equal
  case sublist
  case superlist
  case unequal
}

func classifier(listOne: [Int], listTwo: [Int]) -> Classifier {
  if listOne == listTwo {
    return .equal
  }

  if listOne.windows(ofSize: listTwo.count).contains(where: { $0 == listTwo }) {
    return .superlist
  }

  if listTwo.windows(ofSize: listOne.count).contains(where: { $0 == listOne }) {
    return .sublist
  }

  return .unequal
}

extension Array {
  func windows(ofSize size: Int) -> [[Element]] {
    guard count >= size else { return [] }
    return (0..<(count - size + 1)).map { Array(self[$0..<($0 + size)]) }
  }
}
